src/UI/Component/Entity/Entity.php

```diff
@@ -0,0 +1,130 @@
+     */
+    public function withBlockingAvailabilityConditions(
+        string|array|PropertyListing|StandardLink|Legacy $blocking_conditions
+    ): self;
+
+    /**
+     * Some Properties may be of higher relevance than others, be it as very
+     * significant properties of the entity itself or of greater importance
+     * within the context; you may "feature" them, i.e. put them in a prominent
+     * place.
+     *
+     * @param string|PropertyListing|StandardLink|Legacy|array<string|PropertyListing|StandardLink|Legacy> $featured_props
+     */
+    public function withFeaturedProperties(
+        string|array|PropertyListing|StandardLink|Legacy $featured_props
+    ): self;
+
+    /**
+     * Main Details should provide a quick differentiation or choice on the entity.
+     * "Description" would be one of the most prominent examples.
+     *
+     * @param string|PropertyListing|Legacy|array<PropertyListing|Legacy> $main_details
+     */
+    public function withMainDetails(
+        string|array|PropertyListing|Legacy $main_details
+    ): self;
+
+    /**
+     * Users may directly react to the entity, e.g. comment or tag it.
+     * When there are multiple possible reactions, split them e.g. into more
+     * common ones (here) and less often used ones.
+     * Another way of distinguishing Reactions might be the availability/significance
+     * for everybody in contrast to the current user (e.g. rating vs. my favorite)
+     *
+     * @param array<Glyph|Tag> $prio_reactions
+     */
+    public function withPrioritizedReactions(array $prio_reactions): self;
+
+
+    //Further Areas
+
+    /**
+     * Reactions that are less prominent than Prioritized Reactions go here.
+     * @param array<Glyph|Tag> $reactions
+     */
+    public function withReactions(array $reactions): self;
+
+    /**
+     * Properties that could potentially limit a users access to the object
+     * belong to this group. If they _are_ actually blocking access, then
+     * you should place them into Blocking Availability Conditions.
+     *
+     * @param string|PropertyListing|StandardLink|Legacy|array<PropertyListing|StandardLink|Legacy> $availability
+     */
+    public function withAvailability(
+        string|array|PropertyListing|StandardLink|Legacy $availability
+    ): self;
+
+    /**
+     * Details provide further information about the entity - worth knowing and helpful,
+     * but not as significantly important than Main Details or Featured Properties.
+     *
+     * @param string|PropertyListing|Legacy|array<PropertyListing|Legacy> $details
+     */
+    public function withDetails(
+        string|array|PropertyListing|Legacy $details
+    ): self;
+
+    /**
+     * Actions are the things you can actually _do_ with the entity,
+     * e.g. in context of repository items: view, copy, delete, etc.
+     *
+     * @param Shy[] $actions
+     */
+    public function withActions(array $actions): self;
```

Comment:
```suggestion
    public function withActions(Shy ...$actions): self;
```

will make PHP check our typehint (and allow us to drop `@param` in the doc-block).